zerver/views/registration.py

```diff
@@ -78,11 +78,12 @@ def accounts_register(request):
         # contact support.
         return redirect_to_deactivation_notice()
 
-    try:
-        validate_email_for_realm(realm, email)
-    except ValidationError:
-        return HttpResponseRedirect(reverse('django.contrib.auth.views.login') + '?email=' +
-                                    urllib.parse.quote_plus(email))
+    if not realm_creation:
+        try:
+            validate_email_for_realm(realm, email)
+        except ValidationError:
+            return HttpResponseRedirect(reverse('django.contrib.auth.views.login') + '?email=' +
+                                        urllib.parse.quote_plus(email))
```

Comment:
This line no longer has coverage; I handled it for now with a `#nocoverage`.  I think the right sort of test to cover this is one where a user starts the registration process when their email doesn't need to match a given domain, but then when they're partway through registration, we add a restriction to the domain.  

Can you make sure we add a test for this @hackerkid (or @umairwaheed, if you have time, feel free to claim this).